docs/rules/no-restricted-syntax.md

```diff
@@ -0,0 +1,51 @@
+# Disallow certain syntax (no-restricted-syntax)
+
+Some code bases prefer to not use certain features of JavaScript such as `FunctionExpression` or `WithStatement`. This rule warns on the passed node types.
```

Comment:
Haha, not really wanting you to write docs for me :) Sorry if this implied that. I just asked if you had a specific direction in your head.

I'll make this better ;)